### PR TITLE
fix: enhance and tighten up Vercel adapter warnings

### PR DESCRIPTION
**Pull request by dummdidumm**
- warn when prerender setting makes isr config useless
- don't show cron warning when everything's valid
- allow to set isr to false to clear isr config in leafs (else it's impossible to do so because config is merged at the top level)
- prefixed all warnings with "Warning:" which should help detect the Vercel log dashboard show these in condensed mode where only warnings/errors are shown
- `prerender = 'auto'` routes are now also excempt from further processing if we can determine that this is equal to `prerender = true` (because there are no dynamic routes)

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### CODE REVIEW

File: packages/kit/types/index.d.ts [hunk 1]
```diff
@@ -87,7 +87,7 @@ export interface Builder {
 	config: ValidatedConfig;
 	/** Information about prerendered pages and assets, if any. */
 	prerendered: Prerendered;
-	/** An array of dynamic (not prerendered) routes */
+	/** An array of all routes (including prerendered) */
```
Comment (dummdidumm) [inline packages/kit/types/index.d.ts:90]: drive-by comment fix - I don't think we need a changeset for this, it's okay to just release it as part of the next Kit release without special mention

File: packages/adapter-vercel/index.js [hunk 2]
```diff
@@ -219,6 +226,20 @@ const plugin = function (defaults = {}) {
 				group.routes.push(route);
 			}
 
+			if (ignored_isr.size) {
+				builder.log.warn(
+					`\nWarning: The following routes have an ISR config which is ignored because the route is prerendered:`
```
Comment (dummdidumm) [inline packages/adapter-vercel/index.js:231]: I made this a warning, thought about whether or not this is an error, but I think that would be breaking - maybe in the next major? Or would this be more annoying than helpful if it was an error?

File: packages/adapter-vercel/index.js [hunk 3]
```diff
@@ -572,15 +593,26 @@ function validate_vercel_json(builder, vercel_config) {
...
 	}
+}
 
-	console.log('');
+/** @param {import('@sveltejs/kit').RouteDefinition} route */
+function is_prerendered(route) {
+	return (
+		route.prerender === true ||
+		(route.prerender === 'auto' &&
+			!route.segments.some((segment) => segment.dynamic || segment.rest))
```
Comment (Rich-Harris) [inline packages/adapter-vercel/index.js:614]: `rest` implies `dynamic`, so we can get rid of `|| segment.rest`. i think i'd reverse how the condition is expressed — 'every segment is non-dynamic' is easier to grok than `it's not the case that some segments are dynamic'

```suggestion
			route.segments.every((segment) => !segment.dynamic))
```

File: packages/adapter-vercel/index.js (hunk 2 above)
Comment (Rich-Harris) [reply packages/adapter-vercel/index.js:231]: I think a warning is probably fine, you could have a prerendered leaf within an ISR'd group, for example. we can see if people trip up on it in practice

**Review by Rich-Harris — APPROVED** [review]
made one small suggestion but otherwise LGTM
